zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/appliers/UserTaskAssignmentDeniedApplierTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.appliers;
+
+import io.camunda.zeebe.engine.state.immutable.UserTaskState.LifecycleState;
+import io.camunda.zeebe.engine.state.mutable.MutableProcessingState;
+import io.camunda.zeebe.engine.state.mutable.MutableUserTaskState;
+import io.camunda.zeebe.engine.util.ProcessingStateExtension;
+import io.camunda.zeebe.protocol.impl.record.value.usertask.UserTaskRecord;
+import io.camunda.zeebe.protocol.record.intent.UserTaskIntent;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.api.Assumptions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(ProcessingStateExtension.class)
+public class UserTaskAssignmentDeniedApplierTest {
+
+  /** Injected by {@link ProcessingStateExtension} */
+  private MutableProcessingState processingState;
+
+  /** The class under test. */
+  private UserTaskAssignmentDeniedApplier userTaskAssignmentDeniedApplierApplier;
+
+  /** Used for state assertions. */
+  private MutableUserTaskState userTaskState;
+
+  /** For setting up the state before testing the applier. */
+  private TestSetup testSetup;
+
+  @BeforeEach
+  public void setup() {
+    userTaskAssignmentDeniedApplierApplier = new UserTaskAssignmentDeniedApplier(processingState);
+    userTaskState = processingState.getUserTaskState();
+    testSetup = new TestSetup(processingState);
+  }
+
+  @Test
+  public void shouldRevertAssigneeIfAssignmentDeniedByTaskListener() {
+    // given
+    final var userTaskKey = 1;
+    final var initialAssignee = "initial";
+    final var newAssignee = "changed";
+
+    final var given = new UserTaskRecord().setUserTaskKey(userTaskKey).setAssignee(initialAssignee);
```

Comment:
🔧 With #24233 we're actually changing the data with which user tasks are created if they have an `assignee` defined. We will no longer write these in the `CREATED` event. Instead, separate `ASSIGNING` and `ASSIGNED` events are appended after the `CREATED` event with that property set. While not strictly necessary here, it may be good to already align here as well.

Reply:
Done 👍